extensions/amp-experiment/1.0/mutation/attribute-mutation-default-style.js

```diff
@@ -27,42 +37,82 @@ export class AttributeMutationDefaultStyle {
    * @param {!Array<Element>} elements
    */
   constructor(mutationRecord, elements) {
+    /** @private {!JsonObject} */
     this.mutationRecord_ = mutationRecord;
+
+    /** @private {!Array<Element>} */
     this.elements_ = elements;
+
+    /** @private {!JsonObject} */
+    this.styles_ = dict({});
+
     assertAttributeMutationFormat(this.mutationRecord_);
   }
 
   /** @override */
-  validate() {
+  parseAndValidate() {
     const value = this.mutationRecord_['value'];
-
-    // Do not allow Important or HTML Comments
-    if (value.match(/(!\s*important|<!--)/)) {
+    // First check for !important and <;
+    if (value.match(/(!\s*important|<)/)) {
       return false;
     }
 
-    // Allow Color
-    if (value.match(/^color:\s*#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3});?$/)) {
-      return true;
-    }
+    // Then seperate the style values to pairs in the format "name : value;"
+    // Already guareentee that ['value'] is defined
+    const pairs = value.split(';');
+    for (let i = 0; i < pairs.length; i++) {
+      if (!NON_SPACE_REGEX.test(pairs[i])) {
+        // Note: treat empty string as valid;
+        continue;
+      }
+      // In format of key:value
+      const pair = pairs[i].split(':');
+      if (pair.length != 2) {
+        // more than one :
+        return false;
+      }
 
-    // Allow Background color
-    if (
-      value.match(/^background-color:\s*#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3});?$/)
-    ) {
-      return true;
+      const key = pair[0].trim();
+      if (pair[1] === undefined) {
```

Comment:
would this ever happen?

Reply:
Good point, not anymore after I added the `pair.length !=2` check. I can remove this

Reply:
done